source/common/router/router.cc

```diff
@@ -1229,6 +1229,16 @@ void Filter::onUpstreamHeaders(uint64_t response_code, Http::ResponseHeaderMapPt
     headers->addReferenceKey(Http::Headers::get().SetCookie, header_value);
   }
 
+  // This header is added without checking for config_.suppress_envoy_headers_ to mirror what is
+  // done for upstream requests. Note: this work is done here instead of in
+  // route_entry_->finalizeResponse headers because the attempt_count_ data pertains to the router.
+  // This could be moved, if desired, by charging callbacks_->streamInfo with the attempt count.
+  // Adding the information to the stream info would also be necessary if we wanted this header to
+  // be added on sendLocalReply.
+  if (include_attempt_count_) {
```

Comment:
From a docs perspective it's much simpler to do things this way than explain which local responses get attempt count and which don't, but don't we also really want to know the attempt count when we try the max number of times and fail?  That seems like valuable debug info.

Reply:
Yep, definitely valuable. I erred on the side of the minimal change, but per 3. above I am open to adding the header more widely. I'll go ahead and do this.

Reply:
I was tangling myself a bit with stream info. I realized that we only need this information in the router filter so there is no need to attach this info in the stream info. I think this came out pretty nicely in the modify_headers lambda.